game-frontend/src/components/user-script/ErrorLog.tsx

```diff
@@ -0,0 +1,20 @@
+import React, { FC, useState, useEffect } from 'react';
+import { Hook, Console, Decode } from 'console-feed';
+import { Message } from 'console-feed/lib/definitions/Console';
+
+const ErrorLog: FC = () => {
+  const [logs, setLogs] = useState<Message[]>([]);
+
+  useEffect(() => {
+    Hook(window.console, log => {
+      setLogs([...logs, Decode(log)]);
+    });
+  });
```

Comment:
This change unfortunately doesn't fix the issue 😅 
If you want to test what I mean try this: 
```js
  useEffect(() => {
    Hook(window.console, log => {
      debugger;
      setLogs([...logs, Decode(log)]);
    });
  });
```
Click around and write a manual `console.log` - the breakpoint will be executed multiple times instead of just once.

The reason for this is:
Upon each rerender a new `Hook` listener will be instantiated. So when you start clicking around in the application and the `ErrorLog` component gets rendered - the old `Hook` won't get cleaned up.

P.S.: If you want to you can simply omit the `ErrorLog` and I'll do it as soon as I implement the Script Execution Feedback visualization 😃

Reply:
Thank you for the explanation. So would I do it with a Memo? since I want to only register it as soon as the `ErrorLog` component gets rendered the first time.

Reply:
As discussed: we don't need this atm - so let's drop the `Hook` 👍